src/flow_checks/RareDestination.cpp

```diff
@@ -24,16 +24,89 @@
 
 /* ***************************************************** */
 
+u_int32_t RareDestination::getDestinationHash(Flow *f) {
+
+  u_int32_t hash = 0;
+  Host *dest = f->get_srv_host();
+
+  if (f->isLocalToLocal()) {
+
+    char buf[64];
+
+    if (!dest->isMulticastHost() && dest->isDHCPHost()) {
+      char *mac = dest->getMac()->print(buf,sizeof(buf));
+      hash = Utils::hashString(mac);
+    }
+    else if (dest->isIPv6() || dest->isIPv4()) {
+      char *ip = dest->get_ip()->print(buf,sizeof(buf));
+      hash = Utils::hashString(ip);
+    }
+
+  }
+  else if (f->isLocalToRemote()) {
+    hash = Utils::hashString(f->getFlowServerInfo());
+  }
+
+  return hash;
+}
+
+/* ***************************************************** */
+
 void RareDestination::protocolDetected(Flow *f) {
+
   bool is_rare_destination = false;
 
-  /* TODO: check if this is a real rare destination */
-  if (f->getFlowServerInfo() != NULL) {
-#ifdef TODO_HERE
-    ntop->getTrace()->traceEvent(TRACE_NORMAL, "*** Rare destination %s",
-                                 f->getFlowServerInfo());
-    is_rare_destination = true;
-#endif
+  if(f->getFlowServerInfo() != NULL && f->get_cli_host()->isLocalHost()) {
```

Comment:
What about destination IP? This PR seems to aim at supporting both IPs and hostnames, but I see a check on f->getFlowServerInfo() only